apps/studio/src/server/modules/page/page.router.ts

```diff
@@ -322,45 +322,61 @@ export const pageRouter = router({
           })
         }
 
-        const { meta: oldMeta, ...rest } = fullPage.content
-        const newMeta = JSON.parse(meta)
-
-        const newContent =
-          meta === "" ? { ...rest } : { ...rest, meta: newMeta }
-
-        await updateBlobById(tx, {
-          pageId,
-          content: newContent,
-          siteId,
-        })
-
-        const updatedResource = await tx
-          .updateTable("Resource")
-          .where("Resource.id", "=", String(pageId))
-          .where("Resource.siteId", "=", siteId)
-          .where("Resource.type", "in", ["Page", "CollectionPage"])
-          .set({ title, permalink })
-          .returning([
-            "Resource.id",
-            "Resource.type",
-            "Resource.title",
-            "Resource.permalink",
-            "Resource.draftBlobId",
-          ])
-          .executeTakeFirstOrThrow()
-          .catch((err) => {
-            if (get(err, "code") === "23505") {
-              throw new TRPCError({
-                code: "CONFLICT",
-                message: "A resource with the same permalink already exists",
-              })
-            }
-            throw err
+        const { meta: _oldMeta, ...rest } = fullPage.content
+        const pageMetaSchema = getLayoutMetadataSchema(fullPage.content.layout)
+        const validateFn = ajv.compile(pageMetaSchema)
+        try {
+          const newMeta = JSON.parse(meta) as typeof fullPage.content.meta
+          const isValid = validateFn(newMeta)
```

Comment:
@dcshzj fixed this due to eslint rule calling out JSON.parse returns `any`.
Realised no validation on backend.
Added it here